cli/pcluster/config/cfn_param_types.py

```diff
@@ -1278,3 +1278,26 @@ def refresh(self):
             else:
                 default_volume_size = 20
             self.value = default_volume_size
+
+
+class ClusterResourceBucketCfnParam(CfnParam):
+    """Class to manage cluster_resource_bucket parameter."""
+
+    def from_cfn_params(self, cfn_params):
+        """
+        Initialize parameter value by parsing CFN input parameters.
+
+        :param cfn_params: list of all the CFN parameters, used if "cfn_param_mapping" is specified in the definition
+        """
+        cfn_converter = self.definition.get("cfn_param_mapping", None)
+        if cfn_params:
+            cfn_value = get_cfn_param(cfn_params, cfn_converter) if cfn_converter else "NONE"
+            self.value = (
+                None
```

Comment:
The entire code could be simplified with something like:
```python
if cfn_params and get_cfn_param(cfn_params, "RemoveBucketOnDeletion") != "True":
    # When the value of "RemoveBucketOnDeletion" is not "True" the "cluster_resource_bucket" is created by the user.
    cfn_value = get_cfn_param(cfn_params, cfn_converter)
    self.value = self.get_value_from_string(cfn_value)
```

Reply:
We can avoid the check for the `cfn_converter` because we already know that this parameter has the `cfn_param_mapping` attribute in the mappings. --> we can update the method comment too.